pkg/prow/prow.go

```diff
@@ -223,3 +301,28 @@ func getArtifactsDirectoryPrefix(artifactScanner *ArtifactScanner, prowJobURL, j
 
 	return artifactDirectoryPrefix, nil
 }
+
+func getParentStepName(fullArtifactName, artifactDirectoryPrefix string) (string, error) {
+	// => e.g. [ "", "redhat-appstudio-e2e/artifacts/e2e-report.xml" ]
+	sp := strings.Split(fullArtifactName, artifactDirectoryPrefix)
+	if len(sp) != 2 {
+		return "", fmt.Errorf("cannot determine filepath - object name: %s, object prefix: %s", fullArtifactName, artifactDirectoryPrefix)
+	}
+	parentStepFilePath := sp[1]
+
+	// => e.g. [ "redhat-appstudio-e2e", "artifacts", "e2e-report.xml" ]
+	sp = strings.Split(parentStepFilePath, "/")
+	parentStepName := sp[0]
+
+	return parentStepName, nil
+}
+
+func getFileName(fullArtifactName, artifactDirectoryPrefix string) string {
+	sp := strings.Split(fullArtifactName, artifactDirectoryPrefix)
+	parentStepFilePath := sp[1]
```

Comment:
Would it be possible to get an index out of range error when accessing the `sp` slice? 

And just an idea up for discussion that I'm wondering about myself. Would it be easier to use regex instead of all the string splitting?

Reply:
Thanks for your review, Tomas!
Added a similar check to verify the length of the array beforehand.

> Would it be easier to use regex instead of all the string splitting?

Hmm, I try to avoid regex because it looks more brittle and regex strings are harder to read for me. I prefer to have a string splitting instead because it is easier to know what's going on, especially after checking for errors before we start accessing its content.

But I'm not strictly against it, if we want to move on with regex, then it'd be worth creating a separate ticket for it.

Reply:
Works for me. Thank you